src/common/predictive-text-studio-worker.ts

```diff
@@ -15,4 +18,35 @@ export interface PredictiveTextStudioWorker {
    * @param bcp47Tag
    */
   updateBCP47Tag(bcp47Tag: string): Promise<void>;
+  /**
+   * Generate a KMP file and return a blob of it
+   */
+  generateKMPFromStorage(): Promise<Blob | void>;
+  /**
+   * Take a dictionary source and store it
+   * @param name
+   * @param contents
+   * Return how many words were added
+   */
+  addDictionarySourceToProject(name: string, contents: File): Promise<number>;
+  /**
+   * Register a callback that is called directly before the KMP package is generated
+   * @param callback
+   */
+  onPackageCompileStart(callback: () => void): void;
+  /**
+   * Register a callbcak that is called if the KMP package compilation fails
+   * @param callback
+   */
+  onPackageCompileError(error: Error, callback: (err: Error) => void): void;
+  /**
+   * Register a callback that is called when the KMP package has succesfully compiled
+   * @param callback
+   */
+  onPackageCompileSuccess(kmp: Blob, callback: (kmp: Blob) => void): void;
+  /**
+   * Sets optional and required metadata such as BCP-47, language name, author name, copyright string, etc.
+   * @param metadata
+   */
+  setProjectData(metadata: Readonly<RelevantKmpOptions>): void;
```

Comment:
I like this signature! Though I think it's better to add `Partial` as well:

```suggestion
  setProjectData(metadata: Partial<Readonly<RelevantKmpOptions>>): void;
```